kernel/rtlil.h

```diff
@@ -308,10 +308,14 @@ namespace RTLIL
 		bool operator!=(const char *rhs) const { return strcmp(c_str(), rhs) != 0; }
 
 		char operator[](size_t i) const {
-			const char *p = c_str();
+                        const char *p = c_str();
+#ifndef NDEBUG
```

Comment:
Is checking that a series of bytes aren't zero really that expensive? I feel a little unnerved hiding away consistency checks like this.

Reply:
While this function was a modest contributor to the overall time, it was visible in the profile, and the change made it several times faster / easily inlineable.